bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABridgeHandler.java

```diff
@@ -0,0 +1,281 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.binding.builder.ThingStatusInfoBuilder;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlugwiseHABridgeHandler} class is responsible for handling
+ * commands and status updates for the Plugwise Home Automation bridge.
+ * Extends @{link BaseBridgeHandler}
+ *
+ * @author Bas van Wetten - Initial contribution
+ * @author Leo Siepel - finish initial contribution
+ * 
+ */
+
+@NonNullByDefault
+public class PlugwiseHABridgeHandler extends BaseBridgeHandler {
+
+    // Private Static error messages
+
+    private static final String STATUS_DESCRIPTION_COMMUNICATION_ERROR = "Error communicating with the Plugwise Home Automation controller";
+    private static final String STATUS_DESCRIPTION_TIMEOUT = "Communication timeout while communicating with the Plugwise Home Automation controller";
+    private static final String STATUS_DESCRIPTION_CONFIGURATION_ERROR = "Invalid or missing configuration";
+    private static final String STATUS_DESCRIPTION_INVALID_CREDENTIALS = "Invalid username and/or password - please double-check your configuration";
+    private static final String STATUS_DESCRIPTION_INVALID_HOSTNAME = "Invalid hostname - please double-check your configuration";
+
+    // Private member variables/constants
+
+    private @Nullable PlugwiseHABridgeThingConfig config;
+    private @Nullable GatewayInfo gatewayInfo;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable volatile PlugwiseHAController controller;
+
+    private final HttpClient httpClient;
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseHABridgeHandler.class);
+
+    // Constructor
+
+    public PlugwiseHABridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    // Public methods
+
+    @Override
+    public void initialize() {
+        this.config = getConfig().as(PlugwiseHABridgeThingConfig.class);
+
+        if (this.checkConfig()) {
+            logger.debug("Initializing the Plugwise Home Automation bridge handler with config = {}", this.config);
+            try {
+                this.controller = new PlugwiseHAController(httpClient, config.getHost(), config.getPort(),
+                        config.getUsername(), config.getsmileId());
+                this.controller.start(() -> {
+                    setBridgeProperties();
```

Comment:
If you're making calls to external services in initialize you should schedule the method because initialize is expected to finish fast. And external services might block for some time.